### DISCUSSION TITLE
Exclude thread from GC pause

### DISCUSSION THREAD

**Discussion by domportera** [dotnet/runtime#103330] [Ideas]
Using C#, latency-critical applications can be very difficult to maintain - especially if only a part of the application is latency critical (i.e. an audio engine in a larger game framework). I'd personally love a feature that allows a specified thread to run in a somewhat isolated context where memory is only manually allocated and deallocated (using the GC class, for example) so that the thread is guaranteed to not be paused by the GC for its lifetime.

is this an unreasonable feature to request? would it necessitate making other managed memory inaccessible to this thread (e.g. if the thread is a consumer of audio file paths or DSP parameters)?

Im currently running into this roadblock as i would love to not have to use an entirely different language, marshalling, and/or another process just to deal with realtime audio IO

**Comment by huoyaoyuan** [comment]
See the discussion at #96213

> would it necessitate making other managed memory inaccessible to this thread

It's probably true in current architecture. Note that the "other" is very strong here - basically everything.

**Reply by domportera** [reply]
yeah thats basically what i figured. it makes sense. thank you for the link!
